### PR TITLE
doc: update news

### PR DESCRIPTION
**Pull request by komainu8**

### CODE REVIEW

File: doc/source/news.rst [hunk 1]
```diff
@@ -7,6 +7,94 @@
 News
 ====
 
+.. _release-7.1.0:
```
Comment (kou) [inline doc/source/news.rst]: It should be `.. _release-7-1-0:`. `.` isn't suitable for ID.

File: doc/source/news.rst [hunk 2]
```diff
@@ -7,6 +7,94 @@
...
+  representations(#<nan> and  #i1/0, #-i1/0).
+
+* Supported vector for drilldown calc target.
+
+* Partially supported keyword extraction from regexp search.
+  It enables highlight_html and snippet_html for regexp search.
+  [GitHub#787][Reported by takagi01]
+
+* [bulk] Reduced the number of realloc.
+  grn_bulk_*() API supports it.
```
Comment (kou) [inline doc/source/news.rst]: ``...`` markup should be used for codes. (There are many codes in this change.)

File: doc/source/news.rst [hunk 3]
```diff
@@ -7,6 +7,94 @@
...
+
+* Added new function index_column_source_records.
+  It gets source records of index column.[Patch by Naoya Murakami]
+
+* [select] Supported negative "offset" for "offset + size - limit" >= 0
+
+* Added grn_column_cache.
+  It'll improve performance for getter of fixed size column value.
+
+* [groonga] Added --listen-backlog option.
```
Comment (kou) [inline doc/source/news.rst]: It should be:

```rest
[:doc:`/reference/executables/groonga`]
```

File: doc/source/news.rst [hunk 4]
```diff
@@ -7,6 +7,94 @@
...
+
+* Enabled GRN_II_OVERLAP_TOKEN_SKIP_ENABLE only when its value is "yes".
+
+* Deprecated GRN_NGRAM_TOKENIZER_REMOVE_BLANK_DISABLE.
+  Use GRN_NGRAM_TOKENIZER_REMOVE_BLANK_ENABLE=no instead.
+
+* Added new function index_column_source_records.
+  It gets source records of index column.[Patch by Naoya Murakami]
+
+* [select] Supported negative "offset" for "offset + size - limit" >= 0
```
Comment (kou) [inline doc/source/news.rst]: `select` should be link. Same for other tags.

**Comment by komainu8** [comment]
@kou Thank you for your comments.
I've fixed them.

**Pull request merged**